orion/controllers/semesters/semestersController.go

```diff
@@ -0,0 +1,109 @@
+package semesters
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+
+	"github.com/gin-gonic/gin"
+)
+
+const REGEX_SEMESTER_ID = `^[1-9]\d{3,}_((spring)|(summer)|(fall)|(winter))$`
```

Comment:
Can you add comments above these REGEX expressions to briefly explain what the REGEX is checking?